src/Helpers/Traits/UserInputTrait.php

```diff
@@ -22,4 +22,171 @@ public function styleQuestion(string $question, string $default_value = '', bool
+      ],
+      'uri' => [
+        'description' => "Multisite uri to setup drupal site.",
+        'default_value' => "default",
+        'alias' => "l",
+      ],
+      'yes' => [
+        'description' => "Equivalent to --no-interaction.",
+        'default_value' => 'none',
+        'alias' => "y",
+      ],
+      'no' => [
+        'description' => "Cancels at any confirmation prompt.",
+        'default_value' => 'none',
+      ],
+      'hide-command' => [
+        'description' => "Hide Command. Doesn't show the command executed on terminal.",
+        'default_value' => 'none',
+        'alias' => "hide",
+      ],
+      'display-command' => [
+        'description' => "Display Command. Doesn't show the command executed on terminal.",
+        'default_value' => 'none',
+        'alias' => "d",
+      ],
+      'without-product-info' => [
+        'description' => "Doesn't show the product logo and headline.",
+        'default_value' => "none",
+        'alias' => "wpi",
+      ],
+    ];
+
+  }
+
+  /**
+   * Filter input install options.
+   *
+   * @param array $options
+   *   Install input options.
+   * @param string|null $starterkit
+   *   Starter kit name.
+   *
+   * @return array
+   *   Filtered input options.
+   */
+  public function filterInputOptions(array $options, ?string $starterkit = ''): array {
+    switch ($starterkit) {
+      case 'acquia_cms_headless':
+        if ($options['nextjs-app'] === 'no') {
+          unset($options['nextjs-app']);
+          unset($options['nextjs-app-site-url']);
+          unset($options['nextjs-app-site-name']);
+          unset($options['nextjs-app-env-file']);
+        }
+        unset($options['sitestudio-api-key']);
+        unset($options['sitestudio-org-key']);
+        break;
+
+      case 'acquia_cms_enterprise_low_code':
+        unset($options['nextjs-app']);
+        unset($options['nextjs-app-site-url']);
+        unset($options['nextjs-app-site-name']);
+        break;
+
+      case 'acquia_cms_community':
+        unset($options['nextjs-app']);
+        unset($options['nextjs-app-site-url']);
+        unset($options['nextjs-app-site-name']);
+        unset($options['sitestudio-api-key']);
+        unset($options['sitestudio-org-key']);
+        break;
+
+      default:
+        unset($options['demo-content']);
+        unset($options['content-model']);
+        unset($options['dam-integration']);
+        unset($options['gdpr-integration']);
+        unset($options['nextjs-app']);
+        unset($options['nextjs-app-site-url']);
+        unset($options['nextjs-app-site-name']);
+        unset($options['nextjs-app-env-file']);
+        unset($options['sitestudio-api-key']);
+        unset($options['sitestudio-org-key']);
+        unset($options['gmaps-key']);
+        unset($options['without-product-info']);
+        unset($options['no-interaction']);
+        unset($options['uri']);
+        break;
+    }
+
+    return $options;
+  }
+
```

Comment:
@apathak18 I don't this this is a right implementation? If a user adds their own starter kit and includes a question about the "content model," the default case will be triggered and the option may not be passed correctly. This needs to be addressed for proper implementation.
cc @vishalkhode1 

```suggestion
  public function filterInputOptions(array $options, ?string $starterkit = ''): array {
    // get questions using getStarterkitQuestions($starterKit);
    // add getStarterkitQuestions in trait or in Cli to return questions only related to the selected starterkit.
    $optionsToReturn = [];
    foreach($options as $key => $option){ // e.x. here key is demo-content
      if ($starterkit[$key]){
        $optionsToReturn[$key] = $option;
      }
    }
    return $optionsToReturn; // This will only return the options related to the selected starterkit even the user defined custom starterkit.
  }
  
```